src/QMCWaveFunctions/LCAO/LCAOrbitalSet.cpp

```diff
@@ -586,10 +681,55 @@ void LCAOrbitalSet::mw_evaluateDetRatios(const RefVectorWithLeader<SPOSet>& spo_
 
   mw_evaluateValueVPsImplGEMM(spo_list, vp_list, vp_phi_v);
 
+#if defined(ENABLE_OFFLOAD)
+  const size_t nw             = vp_list.size();
+  auto& invRow_deviceptr_list = spo_leader.mw_mem_handle_.getResource().invRow_deviceptr_list;
+  auto& ratios_buffer         = spo_leader.mw_mem_handle_.getResource().ratios_buffer;
+
+  invRow_deviceptr_list.resize(nVPs);
+  ratios_buffer.resize(nVPs);
+
+  std::vector<size_t> nVP_mw(nw, 0);  // number of VPs per walker
+  std::vector<size_t> iVP0_mw(nw, 0); // first VP idx of each walker
+
+  for (size_t iw = 0; iw < nw; iw++)
+    nVP_mw[iw] = vp_list[iw].getTotalNum();
+  // fill with cumulative number of VPs up to current walker
+  std::exclusive_scan(nVP_mw.begin(), nVP_mw.end(), iVP0_mw.begin(), 0);
+
+  for (size_t iw = 0; iw < nw; iw++)
+  {
+    auto istart = iVP0_mw[iw];
+    auto nvp_i  = nVP_mw[iw];
+    std::fill(invRow_deviceptr_list.begin() + istart, invRow_deviceptr_list.begin() + istart + nvp_i,
+              invRow_ptr_list[iw]);
+    //for (size_t idx = 0; idx< nvp_i; idx++)
+    //  invRow_deviceptr_list[istart+idx] = invRow_ptr_list[iw];
+  }
+  auto* invRow_deviceptr_list_ptr = invRow_deviceptr_list.data();
+  auto* vp_phi_v_ptr              = vp_phi_v.data();
+  auto* ratios_buffer_ptr         = ratios_buffer.data();
+  PRAGMA_OFFLOAD("omp target teams distribute parallel for \
+      map(always,to: invRow_deviceptr_list_ptr[:nVPs]) \
+      map(to: vp_phi_v_ptr[:nVPs*requested_orb_size]) \
+      map(always, from: ratios_buffer_ptr[:nVPs])")
+  for (size_t ivp = 0; ivp < nVPs; ivp++)
+  {
+    ratios_buffer_ptr[ivp] = 0;
+    for (size_t iorb = 0; iorb < requested_orb_size; iorb++)
+      ratios_buffer_ptr[ivp] += vp_phi_v_ptr[ivp * requested_orb_size + iorb] * invRow_deviceptr_list_ptr[ivp][iorb];
+  }
+
+  size_t index = 0;
+  for (size_t iw = 0; iw < nw; iw++)
+    for (size_t iat = 0; iat < vp_list[iw].getTotalNum(); iat++)
+      ratios_list[iw][iat] = ratios_buffer[index++];
+#else
```

Comment:
There's something going wrong here, but I haven't been able to track it down yet.
We get device pointers to the psi inv rows (`invRow_ptr_list`), one per walker.
Each walker has a variable number of VPs, so we need to do nVP[iw] dot products with the correct row of psi_inv (or a single gemv per walker).

As an initial attempt to get this working, I create another list of device pointers to the psi_inv rows (`invRow_deviceptr_list`), but the size of this list is nVPs rather than nwalkers (for each walker, I repeat the psi_inv pointer nVP[iw] times so that I can just do dot products with the same indexing into the `invRow_deviceptr_list` and the first dimensino of the splattered `vp_phi_v`).

The latest issue I'm seeing seems to be a problem with the D2H copy of `ratios_buffer` at the end of the offload region. I added some debug printing [here](https://github.com/kgasperich/qmcpack/compare/lcao-ompblas-simple..kgasperich:qmcpack:lcao-ompblas-debug?expand=1) to make sure that the address where the bad copy is happening is actually the ratios_buffer.

```
ratios_buffer_ptr:    0x5582abf8c500                                                                                                         
ratios_buffer_devptr: 0x7fd80de23800                                                                                                         
"PluginInterface" error: Faliure to copy data from device to host. Pointers: host = 0x00005582abf8c500, device = 0x00007fd80de23800, size = 56: Error in cuMemcpyDtoHAsync: an illegal memory access was encountered
```

In this test, ratios_buffer should contain 7 doubles, so `size=56` makes sense if it's in bytes (but not if it's trying to copy `56*sizeof(double)` bytes)